tensilelite/Tensile/SolutionStructs.py

```diff
@@ -1917,9 +1918,9 @@ def isDirectToVgprDoable(state, tc):
       reject(state, "DirectToVgpr%c does not supports InnerUnroll>1"%(tc))
       return False
 
-    # Reject TLU = UnrollMajorLDS (B only)
-    if tc == 'B' and (state["ProblemType"]["TLUA"] == state["UnrollMajorLDSA"] or state["ProblemType"]["TLUB"] == state["UnrollMajorLDSB"]):
-      reject(state, "DirectToVgpr%c does not supports TLU = UnrollMajorLDS"%(tc))
+    # Reject TLU = UnrollMajorLDS (Either for DTVA/B)
+    if (state["ProblemType"]["TLUA"] == state["UnrollMajorLDSA"]) or (state["ProblemType"]["TLUB"] == state["UnrollMajorLDSB"]):
+      reject(state, "DirectToVgpr does not supports TLUA = UnrollMajorLDSA or TLUB = UnrollMajorLDSB")
       return False
```

Comment:
Remove the previous (B only) assumption. DTVA should also need to reject this.
Otherwise lots of DTVA kernels failed to be generated or validated in such case: either failed in writer stage or validation stage. (All of them are related to unexpected **vgprValuAXXXX**)

Reply:
Thanks for your update.
This change looks OK (you can change to use tc as suggested).
I do not remember why I added tc ==B here.... (this part is copied from Tensile)
We should apply the same logic to both A and B....